src/main/java/in/projecteka/consentmanager/user/PatientsController.java

```diff
@@ -59,22 +72,104 @@
-               : Mono.error(new ClientError(BAD_REQUEST,
-                       new ErrorRepresentation(new Error(INVALID_REQUESTER,
-                               signUpRequests.getError().reduce((left, right) -> format("%s, %s", left, right))))));
+                ? Mono.justOrEmpty(signupService.sessionFrom(token))
+                .flatMap(sessionId -> userService.create(signUpRequests.get(), sessionId)
+                        .zipWith(Mono.just(sessionId))
+                        .flatMap(tuple -> signupService.removeOf(tuple.getT2()).thenReturn(tuple.getT1())))
+                : Mono.error(new ClientError(BAD_REQUEST,
+                new ErrorRepresentation(new Error(INVALID_REQUESTER,
+                        signUpRequests.getError().reduce((left, right) -> format("%s, %s", left, right))))));
+    }
+
+    @PostMapping("/generateotp")
+    @ResponseStatus(CREATED)
+    public Mono<GenerateOtpResponse> generateOtp(@RequestBody GenerateOtpRequest request) {
+        return profileService.profileFor(request.getUsername())
+                .switchIfEmpty(Mono.error(ClientError.userNotFound()))
+                .flatMap(this::getVerfiedIdentifier)
+                .switchIfEmpty(Mono.error(ClientError.unAuthorized()))
+                .flatMap(verifiedIdentifier -> getGenerateOtpResponse(verifiedIdentifier, request.getUsername()))
+                .switchIfEmpty(Mono.error(ClientError.failedToGenerateOtp()));
+    }
+
+    @PostMapping("/verifyotp")
+    public Mono<Token> verifyOtp(@RequestBody OtpVerification request) {
+        return userService.verifyOtp(request);
+    }
+
+    private Mono<GenerateOtpResponse> getGenerateOtpResponse(UserSignUpEnquiry userSignUpEnquiry, String userName) {
+        SignUpSession signUpSession = userService.sendOtpForPasswordChange(userSignUpEnquiry, userName).block();
+        OtpMediumType otpMediumType = getOtpMedium(userSignUpEnquiry);
+        String maskedIdentifier = maskedIdentifier(userSignUpEnquiry);
+
+        if (signUpSession == null || otpMediumType == null || maskedIdentifier == null) {
+            return Mono.empty();
+        }
+
+        return Mono.just(GenerateOtpResponse.builder()
+                .sessionId(signUpSession.getSessionId())
+                .otpMediumValue(maskedIdentifier)
+                .otpMedium(otpMediumType.toString())
+                .expiryInMinutes(userService.getExpiryInMinutes())
+                .build());
+    }
+
+    private String maskedIdentifier(UserSignUpEnquiry userSignUpEnquiry) {
+        String maskedIdentifier = userSignUpEnquiry.getIdentifier();
+        if (userSignUpEnquiry.getIdentifierType().equals(IdentifierType.MOBILE.toString())) {
+            if (maskedIdentifier != null && !maskedIdentifier.equals("")) {
+                return maskedIdentifier.replaceAll("\\d(?=(?:\\D*\\d){4})", "*");
+            }
+        }
+        return null;
```

Comment:
This could be written as 
```
if (!userSignUpEnquiry.getIdentifierType().equals(IdentifierType.MOBILE.toString())) {
            return null;
 }
 if (maskedIdentifier != null && !maskedIdentifier.equals("")) {
           return maskedIdentifier.replaceAll("\\d(?=(?:\\D*\\d){4})", "*");
  }
```

Reply:
Currently, it is only MOBILE as the verified identifier. Going forward it could be email too. This is made generic that way. Returning null if it is not mobile would ruin that.

Reply:
Ok.